mbuild/coordinate_transform.py

```diff
@@ -434,6 +434,26 @@ def translate_to(compound, pos):
     _set_particle_positions(compound, atom_positions)
 
 
+def rotate(compound, theta, rotate_around):
+    """Rotate a compound around an arbitrary vector.
+
+    Parameters
+    ----------
+    compound : mb.Compound
+        The compound being rotated.
+    theta : float
+        The angle by which to rotate the compound, in radians.
+    rotate_around : np.ndarray, shape=(3,), dtype=float
+        The axis about which to rotate the compound.
+
+    """
+    if(np.allclose(rotate_around, 0, atol=1e-16)):
+        raise ValueError('Cannot rotate around a zero vector')
+    atom_positions = compound.xyz_with_ports
+    atom_positions = Rotation(theta, rotate_around).apply_to(atom_positions)
+    _set_particle_positions(compound, atom_positions)
+
+
 def rotate_around_x(compound, theta):
```

Comment:
Do you think you could refactor the `rotate_around_` and `spin_` functions to use the new ones that you wrote? Then you can mark the others with a deprecation warning that says please use `rotate(compound, theta, around=[1, 0, 0])` instead.